### ISSUE TITLE
Output report as JSON or TOML or something else easily parsable

### ISSUE THREAD

**Issue by colindean** [RustSec/cargo-audit#40]
It'd be great to be able to execute something like 

    cargo audit --format json
    cargo audit --json

and get the report in a convenient format:

```json
{
  "database": "https://github.com/RustSec/advisory-db.git",
  "database-count": "10",
  "dependencies-count": "215",
  "vulnerabilities-found": true,
  "vulnerabilities": [
    {
      "ID": "RUSTSEC-2018-0003",
      "Crate": "smallvec",
      "Version": "0.2.1",
      "Date": "2018-07-19",
      "URL": "https://github.com/servo/rust-smallvec/issues/96",
      "Title": "Possible double free during unwinding in SmallVec::insert_many",
      "Solution": "upgrade to: >= 0.6.3 OR ^0.3.4 OR ^0.4.5 OR ^0.5.1"
    },
    {
      "ID": "RUSTSEC-2018-0003",
      "Crate": "smallvec",
      "Version": "0.2.1",
      "Date": "2018-07-19",
      "URL": "https://github.com/servo/rust-smallvec/issues/96",
      "Title": "Possible double free during unwinding in SmallVec::insert_many",
      "Solution": "upgrade to: >= 0.6.3 OR ^0.3.4 OR ^0.4.5 OR ^0.5.1"
    }
  ]
}
```

**Comment by tarcieri** [comment]
Great idea! I think this should be as simple as deriving serde’s `Serialize` and `Deserialize` traits for the `Vulnerabilities` and `Vulnerability` types in the `rustsec` crate and then adding some hooks to serialize them in cargo-audit

**Comment by colindean** [comment]
I started on this and am impressed I got as far as I did! I spent more time compiling than adding the feature!

Where I'm at now, I have the JSON being output. However, all of the status messaging goes to STDOUT instead of STDERR. I want to be able to do `cargo audit --json > report.json`.

I need to move the other stuff to STDERR or just not output it at all (or, include it in the report):

```
$ cargo audit --json
    Fetching advisory database from `https://github.com/RustSec/advisory-db.git`
      Loaded 10 security advisories (from /Users/colin/.cargo/advisory-db)
    Scanning Cargo.lock for vulnerabilities (64 crate dependencies)
error: Vulnerable crates found!
[{"advisory":{"id":"RUSTSEC-2018-0003","package":"smallvec","date":"2018-07-19","patched_versions":[">= 0.6.3","^0.3.4","^0.4.5","^0.5.1"],"unaffected_versions":["< 0.3.2"],"affected_arch":null,"affected_os":null,"aliases":[],"references":[],"keywords":["memory-corruption"],"url":"https://github.com/servo/rust-smallvec/issues/96","title":"Possible double free during unwinding in SmallVec::insert_many","description":"If an iterator passed to `SmallVec::insert_many` panicked in `Iterator::next`,\ndestructors were run during unwinding while the vector was in an inconsistent\nstate, possibly causing a double free (a destructor running on two copies of\nthe same value).\n\nThis is fixed in smallvec 0.6.3 by ensuring that the vector's length is not\nupdated to include moved items until they have been removed from their\noriginal positions.  Items may now be leaked if `Iterator::next` panics, but\nthey will not be dropped more than once.\n\nThank you to @Vurich for reporting this bug.\n"},"package":{"name":"smallvec","version":"0.2.1","source":"registry+https://github.com/rust-lang/crates.io-index","dependencies":null}}]

error: 1 vulnerability found!
```

**Comment by colindean** [comment]
Implemented in https://github.com/RustSec/cargo-audit/pull/41, with output that looks like this when I've manually added a dependency on a vulnerable crate-version within the cargo-audit crate:

```json
{
  "database": {
    "advisory-count": 10
  },
  "lockfile": {
    "dependency-count": 64,
    "path": "Cargo.lock"
  },
  "vulnerabilities": {
    "count": 1,
    "found": true,
    "list": [
      {
        "advisory": {
          "affected_arch": null,
          "affected_os": null,
          "aliases": [],
          "date": "2018-07-19",
          "description": "If an iterator passed to `SmallVec::insert_many` panicked in `Iterator::next`,\ndestructors were run during unwinding while the vector was in an inconsistent\nstate, possibly causing a double free (a destructor running on two copies of\nthe same value).\n\nThis is fixed in smallvec 0.6.3 by ensuring that the vector's length is not\nupdated to include moved items until they have been removed from their\noriginal positions.  Items may now be leaked if `Iterator::next` panics, but\nthey will not be dropped more than once.\n\nThank you to @Vurich for reporting this bug.\n",
          "id": "RUSTSEC-2018-0003",
          "keywords": [
            "memory-corruption"
          ],
          "package": "smallvec",
          "patched_versions": [
            ">= 0.6.3",
            "^0.3.4",
            "^0.4.5",
            "^0.5.1"
          ],
          "references": [],
          "title": "Possible double free during unwinding in SmallVec::insert_many",
          "unaffected_versions": [
            "< 0.3.2"
          ],
          "url": "https://github.com/servo/rust-smallvec/issues/96"
        },
        "package": {
          "dependencies": null,
          "name": "smallvec",
          "source": "registry+https://github.com/rust-lang/crates.io-index",
          "version": "0.2.1"
        }
      }
    ]
  }
}

```
